docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,747 @@
+       // false, the earliest page's address will end in "0001". When it is
+       // true, the earliest page's address will be one more than the
+       // current_page, or "0001" if the current_page is "ffff".
+       bool wrapped = 6;
+   }
+
+   message PropertyPage {
+       message ReportedValue {
+           // The index of the reporter id in reporters field
+           uint32 reporter_index = 1;
+           // Approximately when this value was reported, as a Unix UTC timestamp
+           uint64 timestamp = 2;
+
+           // The type-specific value of the update. Only one of these
+           // fields should be used, and it should match the type
+           // specified for this Property in the RecordType.
+           bytes bytes_value = 11;
+           string string_value = 12;
+           sint64 int_value = 13;
+           float float_value = 14;
+           Location location_value = 15;
+       }
+
+       // The name of the page's associated Property and the record_id of
+       // its associated Record. These are required to distinguish pages
+       // with colliding addresses.
+       string name = 1;
+       string record_id = 2;
+
+       // ReportedValues are sorted first by timestamp, then by reporter_index.
+       repeated ReportedValue reported_values = 4;
+   }
+
+
+Properties and PropertyPages whose addresses collide are stored in
+lists alphabetized by Property name.
+
+.. code-block:: protobuf
+
+   message PropertyContainer {
+       repeated Property entries = 1;
+   }
+
+   message PropertyPageContainer {
+       repeated PropertyPage entries = 1;
+   }
+
+.. _RecordTypes:
+
+Record Types
+------------
+
+In order to validate incoming tracking data, Records are assigned a
+RecordType at creation. A RecordType is a user-defined list of
+PropertySchemas, each of which has a name and data type.
+PropertySchemas may be designated as ``required``. A required Property
+must be initialized with a value at the time of a Record's creation.
+For example, a ``Fish`` type might list ``species`` as required, but
+not ``temperature``, since temperature wouldn't be known until
+measurements were taken. Properties not specified at Record creation
+are initialized as empty lists.
+
+.. code-block:: protobuf
+
+   message PropertySchema {
+       enum DataType {
+           BYTES = 0;
+	   STRING = 1;
+	   INT = 2;
+	   FLOAT = 3;
+	   LOCATION = 4;
+       }
+
+       // The name of the property, e.g. "temperature"
+       string name = 1;
+
+       // The Property's type (int, string, etc.)
+       DataType data_type = 2;
+
+       // A flag indicating whether initial values must be provided for the
+       // Property when a Record is created.
+       bool required = 3;
+   }
+
+
+   message RecordType {
+       // A unique human-readable designation for the RecordType
+       string name = 1;
+
+       repeated PropertySchema properties = 2;
+   }
+
+
+A Record will have all and only the Properties listed in its type, and
```

Comment:
I can't quite parse this first clause ("all and only" is confusing me). Does it mean "A Record has only the Properties listed in its type"?  Or something else?  Also, I'd recommend turning the second clause into a separate sentence (or precede it with a semicolon instead of a comma and remove the "and").  

By the way, I love semicolons; I like to sneak some into each doc I work on.  But now I told you, so feel free exclude semicolons from this doc.